### DISCUSSION TITLE
Corrupt data when trying to hydrate the component.

### DISCUSSION THREAD

**Discussion by hmreumann** [livewire/livewire#4844] [7. Bugs (archived)]
I figured out what was causing a bug in my project and created a repo with it.

[REPO](https://github.com/hmreumann/livewire-zero-in-array-key-bug)

The component is "ZeroInArrayBug"

- Clone
- composer install
- php artisan serve
- follow the instructions and you could see the bug there.

The problem is related to having an array with a key "01" or "10", I'm not sure.

**Comment by joshhanley** [comment]
@hmreumann can you share your component class code here so we can have a look without digging through the repo?

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Thanks!

**Reply by hmreumann** [reply]
@joshhanley yes sure...

```
class ZeroInArrayBug extends Component
{
    public $input = "";
    public $words = [];
    public $combinations = [];

    public function submit()
    {
        $this->words = explode(' ', $this->input);

        foreach ($this->words as $word) {
            foreach ($this->words as $second) {
                $this->combinations[$word.$second] = "This is an element with a key a key: {$word} {$second}";
            }
        }
    }

    public function render()
    {
        return view('livewire.zero-in-array-bug');
    }
}
```

The problem comes when the combinations array contains a key "01" or "10".
